### PR TITLE
TopValues Aggregate

### PR DESCRIPTION
**Pull request by benjismith**
Added the top-values aggregate, with its own builder, alongside the other aggregates.

### CODE REVIEW

File: lib/search.js [hunk 1]
```diff
@@ -98,6 +98,22 @@ SearchBuilder.prototype.aggregate = function (type, path, args) {
 }
 
 /**
+ * Add new 'top_values' aggregate parameter.
+ * @param {string} type
+ * @param {string} path
+ * @param {string} value
```
Comment (housejester) [inline lib/search.js]: Params need to be changed and described.

File: lib/search.js [hunk 2]
```diff
@@ -98,6 +98,22 @@ SearchBuilder.prototype.aggregate = function (type, path, args) {
...
+ * @return {SearchBuilder}
+ */
+ SearchBuilder.prototype.top_values = function (path, offset, limit) {
+  if (typeof(offset) !== "undefined" && typeof(limit) !== "undefined") {
+    return this.aggregate('top_values', path, "offset", offset, "limit", limit);
+  } else if (typeof(offset) === "undefined" || typeof(limit) === "undefined") {
+    throw new Error("offset or limit params must be included together, or not at all");
```
Comment (housejester) [inline lib/search.js]: Should probably use an assert here like the others. Won't need the 'else' since the if returns.

File: test/search.test.js [hunk 3]
```diff
@@ -100,6 +100,8 @@ suite('Search', function () {
     db.newSearchBuilder()
     .collection(users.collection)
     .aggregate('stats', 'value.name')
+    .top_values('value.tags')
+    .top_values('value.tags', 20, 10)
```
Comment (housejester) [inline test/search.test.js]: is this showing 2 aggs for top_values? one for the first 10, the other the next 20?

Comment (housejester) [inline test/search.test.js]: We should have some asserts to show how to handle the response. Though the other aggs don't have that either :cry:

Comment (benjismith) [inline test/search.test.js]: Yes, this is showing 2 different aggs. Though, on second thought, they should have different names. There's no reason anybody would ever request the same field, aggregated twice, with different paging params.

**Comment by housejester** [comment]
:+1:

**Pull request merged**